bootstrap-extensions/src/main/java/de/agilecoders/wicket/extensions/markup/html/bootstrap/form/datetime/AbstractDateTimePickerWithIcon.java

```diff
@@ -130,7 +130,7 @@ protected Component newIcon(String wicketId) {
      * @return icon type
      */
     protected IconType newIconType() {
-        return FontAwesome5IconType.calendar_alt_r;
+        return FontAwesome6IconType.calendar_r;
```

Comment:
Here, we'd have a problem if the user uses dervises from this class (or uses (or with `DatetimePickerIconConfig` below) but chose to consume the other variant via maven coordinates, right? (Regardless if we use the 5 or 6 variant, it could go wrong with two versions)

Reply:
I didn't understand the problem here.

Reply:
Components like this one used to depend on GlyphIcons that were bundled with bootstrap and therefore didn't need an additional dependency. PR #746 in 2018 switched from GlyphIcons to font-awesome icons (at the time to font-awesome-4.7.0, as the only version of font-awesome within wicket-bootstrap. The dependency on org.webjars:font-awesome:4.7.0 was non-optional, so the switch from bundled to non-bundled icons for this and similar components worked flawelessly.

Your initial suggestion for this PR was to make the dependency optional. If a user of wicket-bootstrap would not explicitly depend on font-awesome-6, trying to use the date time picker implementations would not work, as either non font-awesome version would be available at all or only version 5, if that was the users's choice.

The current approach of this PR treats the font-awesome dependency as non-optional, defaulting to version 6. If the user chooses to override the version to keep using version 5, the component selects the appropriate version of the icon.